plugins/ipam/dhcp/lease.go

```diff
@@ -72,29 +72,34 @@ func AcquireLease(clientID, netns, ifName string) (*DHCPLease, error) {
 	log.Printf("%v: acquiring lease", clientID)
 
 	l.wg.Add(1)
-	go ns.WithNetNSPath(netns, true, func(_ *os.File) (e error) {
-		defer l.wg.Done()
-
-		link, err := netlink.LinkByName(ifName)
-		if err != nil {
-			errCh <- fmt.Errorf("error looking up %q", ifName)
-			return
-		}
+	go func() {
+		err := ns.WithNetNSPath(netns, true, func(_ *os.File) (e error) {
+			defer l.wg.Done()
+
+			link, err := netlink.LinkByName(ifName)
+			if err != nil {
+				errCh <- fmt.Errorf("error looking up %q", ifName)
+				return
+			}
 
-		l.link = link
+			l.link = link
 
-		if err = l.acquire(); err != nil {
-			errCh <- err
-			return
-		}
+			if err = l.acquire(); err != nil {
+				errCh <- err
+				return
+			}
 
-		log.Printf("%v: lease acquired, expiration is %v", l.clientID, l.expireTime)
+			log.Printf("%v: lease acquired, expiration is %v", l.clientID, l.expireTime)
 
-		errCh <- nil
+			errCh <- nil
```

Comment:
I guess I can omit the `nil` and just close the channel. But there needs to be something to tell the other goroutine that the lease has been acquired ok.